packages/react-avatar/src/components/Avatar/useAvatarStyles.ts

```diff
@@ -77,277 +77,319 @@ const avatarColor = (theme: Theme, name: keyof GlobalSharedColors) => ({
-  state.image.className = ax(useImageStyles(state), state.image.className);
-  state.label.className = ax(useLabelStyles(state), state.label.className);
-
+  const styles = useStyles();
+  state.className = ax(
+    styles.root,
+    state.size === 20 && styles.size20,
+    state.size === 24 && styles.size24,
+    state.size === 28 && styles.size28,
+    state.size === 32 && styles.size32,
+    state.size === 36 && styles.size36,
+    state.size === 40 && styles.size40,
+    state.size === 48 && styles.size48,
+    state.size === 56 && styles.size56,
+    state.size === 64 && styles.size64,
+    state.size === 72 && styles.size72,
+    state.size === 96 && styles.size96,
+    state.size === 120 && styles.size120,
+    state.size === 128 && styles.size128,
+    state.size < 28 && styles.sizeLessThan28,
+    state.size >= 28 && styles.sizeGreaterEqualThan28,
+    state.size >= 32 && styles.sizeGreaterEqualThan32,
+    state.size >= 48 && styles.sizeGreaterEqualThan48,
+    state.size >= 68 && styles.sizeGreaterEqualThan68,
+    state.size >= 120 && styles.sizeGreaterEqualThan120,
+    state.square && state.size < 28 && styles.squareSizeLessThan28,
+    state.square && state.size >= 28 && styles.squareSizeGreaterEqualThan28,
+    state.square && state.size >= 56 && styles.squareSizeGreaterEqualThan56,
+    state.square && state.size >= 96 && styles.squareSizeGreaterEqualThan96,
+    state.icon !== undefined && styles.icon,
+    state.icon !== undefined && state.size < 28 && styles.iconSizeLessThan28,
+    state.icon !== undefined && state.size >= 28 && styles.iconSizeGreaterEqualThan28,
+    state.icon !== undefined && state.size >= 48 && styles.iconSizeGreaterEqualThan48,
+    state.icon !== undefined && state.size >= 96 && styles.iconSizeGreaterEqualThan96,
+    state.icon !== undefined && state.size >= 120 && styles.iconSizeGreaterEqualThan120,
+    (state.active === 'active' || state.active === 'inactive') && styles.activeInactive,
+    activeRing(state) && styles.activeRing,
+    activeRing(state) && state.size < 48 && styles.activeRingSizeLessThan48,
+    activeRing(state) && state.size >= 48 && styles.activeRingSizeGreaterEqualThan48,
+    activeRing(state) && state.size >= 72 && styles.activeRingSizeGreaterEqualThan72,
+    activeShadow(state) && state.size < 32 && styles.activeShadowSizeLessThan32,
+    activeShadow(state) && state.size >= 32 && styles.activeShadowSizeGreaterEqualThan32,
+    activeShadow(state) && state.size >= 48 && styles.activeShadowSizeGreaterEqualThan48,
+    activeShadow(state) && state.size >= 72 && styles.activeShadowSizeGreaterEqualThan72,
+    activeGlow(state) && state.size < 32 && styles.activeGlowSizeLessThan32,
+    activeGlow(state) && state.size >= 32 && styles.activeGlowSizeGreaterEqualThan32,
+    activeGlow(state) && state.size >= 48 && styles.activeGlowSizeGreaterEqualThan48,
+    activeGlow(state) && state.size >= 72 && styles.activeGlowSizeGreaterEqualThan72,
```

Comment:
I suggest to extract `activeRing`, `activeShadow` and `activeGlow` to be constants inside `useAvatarStyles()` as there is no sense to re-compute them multiple times.

```tsx
const activeRing = /* ... */;
const activeShadow = /* ... */;
const activeGlow = /* ... */;
```

```suggestion
    activeRing && styles.activeRing,
    activeRing && state.size < 48 && styles.activeRingSizeLessThan48,
    activeRing && state.size >= 48 && styles.activeRingSizeGreaterEqualThan48,
    activeRing && state.size >= 72 && styles.activeRingSizeGreaterEqualThan72,
    activeShadow && state.size < 32 && styles.activeShadowSizeLessThan32,
    activeShadow && state.size >= 32 && styles.activeShadowSizeGreaterEqualThan32,
    activeShadow && state.size >= 48 && styles.activeShadowSizeGreaterEqualThan48,
    activeShadow && state.size >= 72 && styles.activeShadowSizeGreaterEqualThan72,
    activeGlow && state.size < 32 && styles.activeGlowSizeLessThan32,
    activeGlow && state.size >= 32 && styles.activeGlowSizeGreaterEqualThan32,
    activeGlow && state.size >= 48 && styles.activeGlowSizeGreaterEqualThan48,
    activeGlow && state.size >= 72 && styles.activeGlowSizeGreaterEqualThan72,
```